quic/connection.nim

```diff
@@ -44,7 +44,8 @@ proc startSending(connection: Connection, remote: TransportAddress) =
     try:
       let datagram = await connection.quic.outgoing.get()
       await connection.udp.sendTo(remote, datagram.data)
-    except TransportError:
+    except TransportError as e:
+      connection.loop.fail(e) # This might need to be revisited
```

Comment:
Is this really necessary for the tests to work? It looks odd to me.

Reply:
Without this, the test hangs forever cause of a deadlock in chronos. I'll explain it better.

Reply:
Adding a brief explanation is a good idea.

Reply:
Added a link to this PR